R/utils.R

```diff
@@ -120,6 +120,7 @@ brackify = function(x, quote=FALSE) {
 eval_with_cols = function(orig_call, all_cols) {
   parent = parent.frame(2L)
   fun_uneval = orig_call[[1L]]
+  stopifnot(length(fun_uneval) == 1)
```

Comment:
IINM we need a regression test here?

Reply:
I'm not sure I understand, can you please clarify/suggest a test?

Reply:
Hmm, actually, in the current PR, this change looks out of place, is it still needed?

Reply:
I made this change to address your comment https://github.com/Rdatatable/data.table/pull/5115#discussion_r1493949195 and actually I'm not sure why it is necessary, so you tell me, haha.

Reply:
:upside_down_face: sorry about that.

I think I was originally worried about usage like `data.table::patterns()` instead of the usual (unqualified) `patterns()`, but I don't think it actually matters here, reading again. We can wait for an actual bug report if there is such case; reverted for now.